rbtcwrapperproxy/RBTCWrapperProxy.sol

```diff
@@ -160,12 +260,80 @@ contract RBTCWrapperProxy{
         
-        msg.sender.transfer(reserveAmount);
+        (bool successOfSendRBTC,) = msg.sender.call.value(reserveAmount)("");
+        require(successOfSendRBTC, "Failed to send RBTC to user");
 
         emit LiquidityRemoved(msg.sender, reserveAmount, _amount);
 
         return reserveAmount;
     }
+
+    /**
+     * @notice 
+     * Before calling this function to remove liquidity, users need approve this contract to be able to spend or transfer their pool tokens
+     *
+     * @dev  
+     * The process:     
+     * 1.Transfers pool tokens to this contract
+     * 2.Calls 'removeLiquidity' on LiquidityPoolConverter contract
+     * 3.Calls 'withdraw' on WRBTC contract in order to unwrap WRBTC to RBTC
+     * 4.Sneds RBTC and/or other reserve tokens to user
+     * 
+     * @param _liquidityPoolConverterAddress    address of LiquidityPoolConverter contract
+     * @param _amount                           amount of pool tokens to burn
+     * @param _reserveTokens                    address of each reserve token. The first element should be the address of WRBTC
+     * @param _reserveMinReturnAmounts          minimum return-amount of each reserve token. The first element should be the minimum return-amount of WRBTC
+     */
+    function removeLiquidityFromV1(
+        address _liquidityPoolConverterAddress,
+        uint256 _amount, 
+        IERC20Token[] memory _reserveTokens, 
+        uint256[] memory _reserveMinReturnAmounts
+    )   
+        public 
+        nonReentrant
+        checkAddress(_liquidityPoolConverterAddress)
+    {
+        require(_amount > 0, "The amount should larger than zero");
+        require(address(_reserveTokens[0]) == wrbtcTokenAddress, "The first reserve token must be WRBTC");
+
+        ILiquidityPoolV1Converter _liquidityPoolConverter = ILiquidityPoolV1Converter(_liquidityPoolConverterAddress);
+        ISmartToken _poolToken = ISmartToken(address(_liquidityPoolConverter.token()));
+
+        bool successOfTransferFrom = _poolToken.transferFrom(msg.sender, address(this), _amount);
+        require(successOfTransferFrom);
+
+        uint256 lengthOfToken = _reserveTokens.length;
+        IERC20Token reserveToken;
+        uint256[] memory reserveAmountBefore = new uint256[](lengthOfToken);
+        for (uint256 i = 0; i < lengthOfToken; i++) {
+            reserveToken = _reserveTokens[i];
+            reserveAmountBefore[i] = reserveToken.balanceOf(address(this));
+        }
+
+        _liquidityPoolConverter.removeLiquidity(_amount, _reserveTokens, _reserveMinReturnAmounts);
+
+        uint256 wrbtcAmount = _reserveTokens[0].balanceOf(address(this)).sub(reserveAmountBefore[0]);
+        require(wrbtcAmount >= _reserveMinReturnAmounts[0], "ERR_ZERO_TARGET_AMOUNT");
+        IWrbtcERC20(wrbtcTokenAddress).withdraw(wrbtcAmount);
+        msg.sender.transfer(wrbtcAmount);
+        emit LiquidityRemovedFromV1(msg.sender, wrbtcTokenAddress, wrbtcAmount);
```

Comment:
you can move these lines down to the very end of the function.
then it will follow CEI pattern, and it will become safe to replace `.transfer` with `.call`